### DISCUSSION TITLE
Adopt precedence rules in baptism and interpretation

### DISCUSSION THREAD

**Discussion by ppKrauss** [frictionlessdata/frictionlessdata.io#867] [General]
(this is an extension of hierarchical principle frictionlessdata/frictionlessdata.io#852)

To avoid conflics and long discussions during cration of new "terms": Frictionlessdata *datatype names* or, in a specific dataset, *column names*, and the best choice for URL of `rdfType`.

Illustrating motivations:

* during [tests of implementation at `frictionlessdata/goodtables.io`](https://github.com/frictionlessdata/goodtables.io/issues/228) I perceived the problem with "`gyear` vs `year`"... In fact, there are a [lot of discussions about](https://github.com/frictionlessdata/specs/issues?utf8=%E2%9C%93&q=is%3Aissue%20gyear%20). How to "reduce frictionless"?

* during construction of datasets at [Brazilian datasets](https://github.com/datasets-br) ([example](https://github.com/datasets-br/state-codes)) I need clues to column name definitions, some "best practices guide for column names"...  But looking at  [Core Datasets](https://github.com/datasets) I see "the caos", no clues.

* The `rdfType` will be  welcomed in all projects, but the URL choice can't be random... We need some "best practices guide for URL choice in `rdfType` descriptors".

## Suggesting simple rules

1. *All derivative inherits from its partents assumptions and interpretations.* <br/>**Example-1** (structural derivative): in the case of *year*, it is a derivative of *date*  that already have a datatype, [datetime](http://specs.frictionlessdata.io/table-schema/#datetime), an ISO  format `year-mm-dd`. So, by ISO8601 interpretation, *year* is a "gregorian year", and it is a syntatic component of *datetime* (we can suppose precedence from the components with relation from another external standards). Same with *yearmonth*, is a `YYYY-MM` syntatic component. As "gregorian" is inherited, we [not need repeat](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) as "g"; so "year" is prefered from "gyear"; etc.  <br/>**Example-2** (semantic derivative): person's full-name is a property of  person, so if the "adopted standard" was SchemaOrg's [class `Person`](https://schema.org/Person), the SchemaOrg's recommendation is [`name`](https://schema.org/name) (not ex. "fullName"). For split name into parts, there are also reference standards, as [`familyName`](https://schema.org/familyName), `givenName`, etc. Syntax and interpretation can be inherit.

2. *Adopt precedence order and "reference vocabularies" for `rdfType` and semantics.* <br>**Example**: adopt SchemaOrg class-property when it exist, else (when exist stable one) [Wikidata](https://www.wikidata.org), else... [LOV](https://lov.okfn.org/dataset/lov/) sequence.

**Comment by pwalsh** [comment]
Hi @ppKrauss I'm not ignoring this, but I'm focused on finalising v1 before I address potential changes for subsequent versions. Thanks for your patience.
